src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectOptionsProcessor.cs

```diff
@@ -70,26 +63,22 @@ private bool ReparseCommandLineIfChanged_NoLock(ImmutableArray<string> arguments
 
         _commandLineChecksum = checksum;
 
-        // Dispose the existing stored command-line and then persist the new one so we can
-        // recover it later.  Only bother persisting things if we have a non-empty string.
-
-        _commandLineStorageHandle = null;
-        if (!arguments.IsEmpty)
-        {
-            using var stream = SerializableBytes.CreateWritableStream();
-            using var writer = new StreamWriter(stream);
+        ReparseCommandLine_NoLock(arguments);
 
-            foreach (var value in arguments)
-                writer.WriteLine(value);
+        // Only bother storing the command line if there is an effective ruleset, as that may
+        // require a later reparse using it.
```

Comment:
Probably don't need a comment here since the comment isn't entirely accurate -- if the method is overriden the policy stated isn't always right.